### ISSUE TITLE
MongoError: "Invalid response header"

### ISSUE THREAD

**Issue by kaldaf** [denodrivers/mongo#342]
Hi, I'm trying connect to MongoDb Atlas.

On localhost everything is OK, but when I deploy project on Deno Deploy it throw this error: 
`Error: MongoError: "Invalid response header"
    at WireProtocol.receive (https://deno.land/x/mongo@v0.29.3/src/protocol/protocol.ts:77:38)`

![err](https://user-images.githubusercontent.com/50804954/162687584-4ef5891d-e564-466a-ab39-ca991d7f21fc.JPG)

I already have on Mongo network access with Deno Deploy IP.

This is my usage of your module:
_mongodb.ts_
```typescript
import { MongoClient } from "./deps.ts";
import "https://deno.land/x/dotenv@v3.2.0/load.ts";
const client = new MongoClient();

const MONGO_URI = Deno.env.get("MONGO_URI");
const db = await client.connect(MONGO_URI ? MONGO_URI : "");

export default db;
```

_app.ts_
```typescript
import db from "../mongodb.ts";
const usersCollection = db.collection<IUserModel>("users");

export const authLogin = async(ctx: Context) =>
{
  if (!ctx.request.hasBody) ctx.throw(Status.BadRequest, "Bad Request");

  const model = await ctx.request.body().value;
  if (!model) ctx.throw(Status.BadRequest, "Bad Request");

  const user = await usersCollection.findOne(
    { email: model.email },
    { noCursorTimeout: false }
  );
 ... // more code
}
```

As I say on localhost, everything works, but when I deploy it, it throw Crash Report.

Any idea how to solve this?

**Comment by lucsoft** [comment]
are you sure your ENVs are set?

**Comment by kaldaf** [comment]
It's temporary, I also had a hard URI there, same error.

**Comment by erfanium** [comment]
@kaldaf You mean you can access MongoDB Atlas via local system but not Deno Deploy?
BTW https://mongo-demo.deno.dev/ is a demo with MongoDB Atlas free M0 tier deployed on Deno Deploy and works well

**Comment by kaldaf** [comment]
@erfanium Yes

I switched it to:

```typescript
import { MongoClient } from "./deps.ts";
const client = new MongoClient();

const db = await client.connect("mongodb+srv://NAME:PASS@CLUSTER.xyywq.mongodb.net/DB?authMechanism=SCRAM-SHA-1");

export default db;
```

And same err:
`isolate start time: 2 milliseconds
1 Error: MongoError: "Invalid response header" at WireProtocol.receive (https://deno.land/x/mongo@v0.29.3/src/protocol/protocol.ts:77:38)`

**Comment by kaldaf** [comment]
And about mongo atlas:
VERSION - 5.0.6
REGION - AZURE Netherlands (westeurope)
CLUSTER TIER - M0 Sandbox (General)

Maybe problem with Azure?

**Comment by lucsoft** [comment]
We have problems with MongoDB Atlas, if i have more time i wanna look into better Altas support but my finals are less then a month away after that i have more time (i think)

**Comment by erfanium** [comment]
![Screenshot from 2022-04-11 13-43-39](https://user-images.githubusercontent.com/47688578/162704743-cdc184bd-4a96-40fd-8b4c-df2829cc3c98.png)

This should works (the database behind https://mongo-demo.deno.dev)

**Comment by BrunoBernardino** [comment]
I also have this problem sporadically with M2 in Cloud Atlas. I think it's when the primary server in the cluster switches, but basically for a few minutes this error shows up and blocks everything.

If there's something I can do to improve support on this, I'm happy to help.

**Comment by 0xxFF** [comment]
I also had the same problem. I solved it by allowing network access from 0.0.0.0/0 (I only had my local ip set)

**Comment by Torjoy81** [comment]
Bro go to netwrok access add ip address click Allow from anywhere
